### PR TITLE
content graph relationships and dependency fixes

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
* Relationship hash value wasn't unique, causing some relationships not to be included
* Fixed `to_id_set` method to have tha pack folder instead of display_name
* Fix dependency calculation 

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

**Review by dantavori — APPROVED** [review]

File: demisto_sdk/commands/content_graph/objects/content_item.py [hunk 1]
```diff
@@ -28,25 +28,25 @@ class ContentItem(BaseContent):
 
     @property
     def in_pack(self) -> Optional["Pack"]:
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/objects/content_item.py:30]: need docstrings and/or comments.

File: demisto_sdk/commands/content_graph/objects/content_item.py [hunk 2]
```diff
@@ -28,25 +28,25 @@ class ContentItem(BaseContent):
...
-            if r.relationship_type == RelationshipType.IN_PACK:
-                return r.content_item  # type: ignore[return-value]
-        return None
+        in_pack = self.relationships_data[RelationshipType.IN_PACK]
+        if not in_pack:
+            return None
+        return next(iter(in_pack)).content_item  # type: ignore[return-value]
 
     @property
     def uses(self) -> List["RelationshipData"]:
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/objects/content_item.py:37]: need docstrings and/or comments.

File: demisto_sdk/commands/content_graph/objects/content_item.py [hunk 3]
```diff
@@ -28,25 +28,25 @@ class ContentItem(BaseContent):
...
         return [
             r
-            for r in self.relationships_data
-            if r.relationship_type == RelationshipType.USES and r.content_item == r.target
+            for r in self.relationships_data[RelationshipType.USES]
+            if r.content_item == r.target
         ]
 
     @property
     def tested_by(self) -> List["TestPlaybook"]:
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/objects/content_item.py:45]: need docstrings and/or comments.

**Review by tkatzir — APPROVED** [review]
Approved, with minor comments.

**Pull request merged**